### PR TITLE
move collection page entry to pill on web + mobile

### PR DESCRIPTION
**Pull request by kaitoo1**
### Summary of Changes
This PR moves the link to the collection page into a pill on web + mobile

- [x] pill links to collection page
- [x] if no collection name, display "Unknown Collection"
- [x] truncate collection name if it's longer than the container
- [x] dark mode is good on mobile

### Demo or Before/After Pics
Before - web
CleanShot 2023-08-31 at 23 07 37@2x

Before - mobile
CleanShot 2023-08-31 at 23 08 52@2x

After - web
CleanShot 2023-08-31 at 23 03 31@2x

After - mobile 
CleanShot 2023-08-31 at 23 06 06@2x

After- mobile dark
CleanShot 2023-08-31 at 23 06 12@2x

After - web long text
CleanShot 2023-08-31 at 23 04 26@2x

After- mobile long text
CleanShot 2023-08-31 at 23 07 17@2x

- If this is a new feature, include screenshots or recordings of the feature in action.
- If this PR makes visual changes, include before and after screenshots. You can type `/table` to quickly come up with a before/after table.

### Edge Cases
- long collection name
- no collection name 

### Testing Steps
Go to any post and look for the collection name

### Checklist

Please make sure to review and check all of the following:

- [x] I've tested the changes and all tests pass.
- [x] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [x] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/mobile/src/components/Feed/Posts/PostListItem.tsx [hunk 1]
```diff
@@ -55,6 +66,15 @@ export function PostListItem({ feedPostRef, queryRef }: Props) {
...
 
   const firstToken = feedPost.tokens?.[0] || null;
+  const community = firstToken?.community ?? null;
+
+  const handleCommunityPress = useCallback(() => {
+    if (!community || !community.contractAddress) return;
+    navigation.push('Community', {
+      contractAddress: community.contractAddress?.address ?? '',
+      chain: community.contractAddress?.chain ?? '',
+    });
```
Comment (Robinnnnn) [inline apps/mobile/src/components/Feed/Posts/PostListItem.tsx:76]: nit: should we also check for the existence of `.address` and `.chain` in the `if` statement and return early if they don't exist? doesn't make sense to push to a community page with empty string params right?

File: apps/web/src/components/Feed/Posts/PostCommunityPill.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    `,
+    postRef
+  );
+
+  const token = post.tokens && post.tokens[0];
+
+  const track = useTrack();
+
+  const handleClick = useCallback(() => {
+    track('Clicked Post COmmunity Pill', { community: token?.community?.name });
```
Comment (Robinnnnn) [inline apps/web/src/components/Feed/Posts/PostCommunityPill.tsx:35]: `COmmunity` -> `Community`

**Review by Robinnnnn — APPROVED** [review]

File: apps/web/src/components/Feed/Posts/PostCommunityPill.tsx (hunk 2 above)
Comment (kaitoo1) [reply apps/web/src/components/Feed/Posts/PostCommunityPill.tsx:35]: wHaTS WroNg wITh iT

File: apps/mobile/src/components/Feed/Posts/PostListItem.tsx (hunk 1 above)
Comment (kaitoo1) [reply apps/mobile/src/components/Feed/Posts/PostListItem.tsx]: good call! inverted the condition so we only push if address and chain are present

File: apps/web/src/components/Feed/Posts/PostCommunityPill.tsx (hunk 2 above)
Comment (kaitoo1) [reply apps/web/src/components/Feed/Posts/PostCommunityPill.tsx]: (fixed)

**Pull request merged**